frontends/mit-open/src/page-components/ResourceCarousel/ResourceCarousel.tsx

```diff
@@ -0,0 +1,298 @@
+    default:
+      // @ts-expect-error This will always be an error if the switch statement
+      // is exhaustive since dataConfig will have type `never`
+      throw new Error(`Unknown data type: ${dataConfig.type}`)
+  }
+}
+
+type ResourceCarouselProps = {
+  config: TabConfig[]
+  title: string
+}
+
+const HeaderRow = styled.div(({ theme }) => ({
+  display: "flex",
+  flexWrap: "wrap",
+  alignItems: "center",
+  gap: "16px",
+  marginBottom: "24px",
+  [theme.breakpoints.down("sm")]: {
+    alignItems: "flex-start",
+    flexDirection: "column",
+  },
+}))
+const ControlsContainer = styled.div({
+  display: "flex",
+  flex: 1,
+  minWidth: "0px",
+  maxWidth: "100%",
+  justifyContent: "space-between",
+})
+
+const StyledTabPanel = styled(TabPanel)({
+  paddingTop: "0px",
+  paddingLeft: "0px",
+  paddingRight: "0px",
+})
+
+const ButtonsContainer = styled.div(({ theme }) => ({
+  display: "flex",
+  gap: "8px",
+  [theme.breakpoints.down("sm")]: {
+    display: "none",
+  },
+}))
+
+const TabsList = styled(TabButtonList)({
+  ".MuiTabScrollButton-root.Mui-disabled": {
+    display: "none",
+  },
+})
+
+type ContentProps = {
+  resources: LearningResource[]
+  isLoading?: boolean
+  tabConfig: TabConfig
+}
+
+type PanelChildrenProps = {
+  config: TabConfig[]
+  children: (props: ContentProps) => React.ReactNode
+}
+const PanelChildren: React.FC<PanelChildrenProps> = ({ config, children }) => {
+  if (config.length === 1) {
+    return (
+      <DataPanel dataConfig={config[0].data}>
+        {({ resources, isLoading }) =>
+          children({ resources, isLoading, tabConfig: config[0] })
+        }
+      </DataPanel>
+    )
+  }
+  return (
+    <>
+      {config.map((tabConfig, index) => (
+        <StyledTabPanel key={index} value={index.toString()}>
+          <DataPanel dataConfig={tabConfig.data}>
+            {({ resources, isLoading }) =>
+              children({
+                resources,
+                isLoading,
+                tabConfig,
+              })
+            }
+          </DataPanel>
+        </StyledTabPanel>
+      ))}
+    </>
+  )
+}
+
+const MobileOverflow = styled.div(({ theme }) => ({
+  /**
+   * On mobile screens, the carousel is supposed to overflow the main content
+   * so its right edge is flush with screen.
+   *
+   * The mobile content margin is 16px, so we add that to its width.
+   */
+  [theme.breakpoints.down("sm")]: {
+    width: "calc(100% + 16px)",
+  },
+}))
+
+/**
+ * A tabbed carousel that fetches resources based on the configuration provided.
+ *  - each TabConfig generates a tab + tabpanel that pulls data from an API based
+ *    on the config
+ *  - data is lazily when the tabpanel first becomes visible
+ *
+ * For now, this is a carousel of learning resource cards, to be moved out if/when it is needed for other items.
+ *
+ * If there is only one tab, the carousel will not have tabs, and will just show
+ * the content.
```

Comment:
We aren't using a "single-tab" version yet, but I think it will be useful soon, e.g., on the profile page https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=4304-55066&m=dev for "Top picks for you".